tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/util/BCompileUtil.java

```diff
@@ -141,30 +142,49 @@ private static void runInit(BLangPackage bLangPackage, JBallerinaInMemoryClassLo
         String funcName = cleanupFunctionName(name);
         try {
             final Method method = initClazz.getDeclaredMethod(funcName, Strand.class);
-            Scheduler scheduler = new Scheduler(4, false);
+            Scheduler scheduler = scheduler1;
             //TODO fix following method invoke to scheduler.schedule()
-            method.invoke(null, new Strand(scheduler));
-        } catch (InvocationTargetException e) {
-            Throwable t = e.getTargetException();
-            if (t instanceof org.ballerinalang.jvm.util.exceptions.BLangRuntimeException) {
-                throw new org.ballerinalang.util.exceptions.BLangRuntimeException(t.getMessage());
-            }
-            if (t instanceof org.ballerinalang.jvm.util.exceptions.BallerinaConnectorException) {
-                throw new org.ballerinalang.util.exceptions.BLangRuntimeException(t.getMessage());
-            }
-            if (t instanceof ErrorValue) {
-                throw new org.ballerinalang.util.exceptions
-                        .BLangRuntimeException("error: " + ((ErrorValue) t).getPrintableStackTrace());
+            Function<Object[], Object> func = objects -> {
+                try {
+                    return method.invoke(null, objects[0]);
+                } catch (IllegalAccessException e) {
+                    e.printStackTrace();
+                } catch (InvocationTargetException e) {
+                    e.printStackTrace();
+                }
+                return null;
+            };
+            final FutureValue out = scheduler.schedule(new Object[1], func, null, null, null);
+            scheduler.start();
+            if (out.panic != null) {
+                throw new RuntimeException("Error in init/start", out.panic);
             }
-            throw new RuntimeException("Error while invoking function '" + funcName + "'", e);
+            //method.invoke(null, new Strand(scheduler));
+            //        } catch (InvocationTargetException e) {
+            //            Throwable t = e.getTargetException();
+            //            if (t instanceof org.ballerinalang.jvm.util.exceptions.BLangRuntimeException) {
+            //                throw new org.ballerinalang.util.exceptions.BLangRuntimeException(t.getMessage());
+            //            }
+            //            if (t instanceof org.ballerinalang.jvm.util.exceptions.BallerinaConnectorException) {
+            //                throw new org.ballerinalang.util.exceptions.BLangRuntimeException(t.getMessage());
+            //            }
+            //            if (t instanceof ErrorValue) {
+            //                throw new org.ballerinalang.util.exceptions
+            //                        .BLangRuntimeException("error: " + ((ErrorValue) t).getPrintableStackTrace());
+            //            }
+            //            throw new RuntimeException("Error while invoking function '" + funcName + "'", e);
```

Comment:
Shall we remove commented code as it can be viewed in the commit history in github?